packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -171,13 +170,8 @@ export function makeVatKeeper(
   }
 
   function getIncarnationNumber() {
-    return Number(getRequired(`${vatID}.incarnationNumber`));
-  }
-
-  function incIncarnationNumber() {
-    const newIncarnationNumber = getIncarnationNumber() + 1;
-    kvStore.set(`${vatID}.incarnationNumber`, `${newIncarnationNumber}`);
-    return newIncarnationNumber;
+    const { incarnation } = transcriptStore.getCurrentSpanBounds(vatID);
```

Comment:
nit: prefer `return transcriptStore.getCurrentSpanBounds(vatID).incarnation;`, if it all fits on a single line